cppcheck_junit.py

```diff
@@ -128,14 +156,26 @@ def generate_test_suite(errors: Dict[str, List[CppcheckError]]) -> ElementTree.E
             time=str(1),
         )
         for error in errors:
-            ElementTree.SubElement(
+            error_element = ElementTree.SubElement(
                 test_case,
                 "error",
-                type="",
+                type=error.severity,
                 file=os.path.relpath(error.file) if error.file else "",
-                line=str(error.line),
-                message=f"{error.line}: ({error.severity}) {error.message}",
+                message=f"{error.message}",
             )
+            if len(error.locations) == 0:
+                error_element.text = error.verbose
+            elif len(error.locations) == 1 and error.locations[0].info == "":
+                location = error.locations[0]
+                file = os.path.relpath(location.file) if location.file else ""
+                error_element.text = f"{file}:{location.line}:{location.column}: {error.verbose}"
+            else:
+                error_element.text = error.verbose
+                for location in error.locations:
+                    file = os.path.relpath(location.file) if location.file else ""
+                    error_element.text += (
+                        f"\n{file}:{location.line}:{location.column}: {location.info}"
+                    )
```

Comment:
3 cases adds the rest of the info in the `text` part of the `error` element.
- In the simplest case, just put the `verbose` content.
- If you've a specific `location`, but no extra `info` add `file:line:column: verbose`
- Otherwise, Add `verbose` first, then `file:line:column: info` from all `location`s